### PR TITLE
Validate unique sample names

### PR DESCRIPTION
**Pull request by seallard**
Add validation that sample names are unique per case for tomte order.

### CODE REVIEW

File: cg/services/order_validation_service/workflows/tomte/validation/inter_field/utils.py [hunk 1]
```diff
@@ -47,3 +51,14 @@ def _get_sample_well_map(plate_samples_with_cases: list[tuple[TomteSample, Tomte
             sample_well_map[sample.well_position] = []
         sample_well_map[sample.well_position].append((sample, case))
     return sample_well_map
+
+
+def get_duplicate_sample_names(case: TomteCase) -> list[str]:
+    sample_names = [sample.name for sample in case.samples]
+    count = Counter(sample_names)
+    return list({name for name, freq in count.items() if freq > 1})
```
Comment (islean) [inline cg/services/order_validation_service/workflows/tomte/validation/inter_field/utils.py:59]:
```suggestion
    return [name for name, freq in count.items() if freq > 1]
```

**Review by islean — APPROVED** [review]
Looks great!

File: cg/services/order_validation_service/workflows/tomte/validation/inter_field/utils.py (hunk 1 above)
Comment (islean) [reply cg/services/order_validation_service/workflows/tomte/validation/inter_field/utils.py:59]: or why the set conversion?

**Review by ChrOertlin — APPROVED** [review]
Clean

File: cg/services/order_validation_service/models/errors.py [hunk 2]
```diff
@@ -53,3 +53,8 @@ class OrderNameRequiredError(OrderError):
 class OccupiedWellError(CaseSampleError):
     field: str = "well_position"
     message: str = "Well is already occupied"
+
+
+class ReusedSampleNameError(CaseSampleError):
```
[the remark this answers is not in the document]
Comment (seallard) [reply cg/services/order_validation_service/models/errors.py]: https://github.com/Clinical-Genomics/cg/pull/3480/commits/fa171d855675cc3d4241b1acfa1e3992d0fc0b44

**Pull request merged**